src/main/java/net/minestom/server/instance/EntityTrackerImpl.java

```diff
@@ -106,17 +138,23 @@ public void remove(@NotNull T removed) {
         }
     }
 
+    @Override
+    public void changeUuid(@NotNull Entity entity, UUID oldUuid) {
```

Comment:
this is not thread safe and missing an annotation, but honestly we should just remove the setUuuid method from entity and this functionality and make it final. It doesn't make sense to change an entity uuid after creating it I dont think.

Reply:
The only use within Minestom of setUuid within Minestom is transitionLoginToConfig
The commit message said thread-safety, I meant safer than register/unregister.

I agree with removing setUwuid, but I think it's beyond the scope of this PR.